frontend/src/app/modules/matches/matches-computations.js

```diff
@@ -68,3 +79,71 @@ export const computeWinRatio = (playerId, playerMatches) => {
   const wonMatchesCount = playerMatches.filter(match => didPlayerWin(playerId, match)).length
   return (playerMatches.length > 0) ? (wonMatchesCount / playerMatches.length) : 0
 }
+
+export const computeWins = (playerId, playerMatches) =>
+  playerMatches.filter(match => didPlayerWin(playerId, match)).length
+
+const findWithComparator = (comparator, arr, getField, baseScore) =>
+  arr.reduce((acc, node) =>
+    comparator(acc.score, getField(node))
+      ? acc
+      : { value: node.value, score: getField(node) }
+  , { score: baseScore })
+
+export const findMin = (arr, getField = node => node.score) =>
+  findWithComparator((e1, e2) => e1 <= e2, arr, getField, Infinity)
+
+export const findMax = (arr, getField = node => node.score) =>
+  findWithComparator((e1, e2) => e1 >= e2, arr, getField, -Infinity)
+
+export const computeDays = matchChanges => {
+  const daysMap = new Map()
+
+  for (const match of matchChanges) {
+    const day = match.date.toLocaleDateString()
+    const score = Number(match.ratingChangeString)
+    daysMap.set(day, !daysMap.has(day)
+      ? {
+        value: day,
+        score,
+      }
+      : {
+        ...daysMap.get(day),
+        score: daysMap.get(day).score + score,
+      })
+  }
+  return matchChanges.length
+    ? Array.from(daysMap.values())
+    : DEFAULT_DAYS_STATISTICS
+}
+
+export const computeTeammateStats = (playerId, playerMatches) =>
+  computePlayerStats(playerId, playerMatches, getTeamMates) || {}
+
+export const computeOpponentsStats = (playerId, playerMatches) =>
+  computePlayerStats(playerId, playerMatches, getOpponents) || {}
+
+const computePlayerStats = (playerId, playerMatches, playersProvider) => {
+  const playersMap = new Map()
+
+  for (const match of playerMatches) {
+    const players = playersProvider(playerId, match)
+    players.forEach(player =>
+      playersMap.set(player.id, !playersMap.has(player.id)
```

Comment:
i see that you are fond of negating first :D well I guess there are two kinds of people in the world :D

Reply:
De-negated [here](https://github.com/salsita/foosball-rating/pull/114/commits/54023373857edbdd2d03d427c0b799b98214f08c#diff-304dca28d92d292252e86b7e83fc1569)